vllm/engine/arg_utils.py

```diff
@@ -467,6 +468,13 @@ def add_cli_args(
             'draft model. Sequences over this length will skip '
             'speculation.')
 
+        parser.add_argument(
+            '--speculative-disable-queue-size',
+            type=int,
+            default=EngineArgs.speculative_disable_queue_size,
+            help='Disable speculative decoding for new incoming requests '
+            'if the queue size is larger than this value.')
```

Comment:
Can we pick a better name? It isn't clear from the name that this refers to running queue. And even if it did, `running_queue` is only explanatory for those familiar with the scheduler nomenclature. Perhaps we can use batch size?

Reply:
I also feel this name is not really ideal, and we (@LiuXiaoxuanPKU) plan to change it to be more consistent with other arguments in DSD once it's out, but I'm open to any suggest as I'm always bad at naming.

We cannot use batch size, because it indicates the size of the current batch. The information we need here is more like a system loading, so we instead need the size of running queue.

Reply:
`running queue == scheduled requests`

https://github.com/vllm-project/vllm/blob/19cb4716ee700e5d8baa64d7cf14fb5da3737f6d/vllm/core/scheduler.py#L286-L291

Reply:
Hmm I was looking at https://github.com/vllm-project/vllm/blob/main/vllm/core/scheduler.py#L774 which seems cover not only the scheduled requests, or maybe I misunderstood...

Reply:
yeah we could call it num_running_or_waiting